From e2e9da35d4a598490b73da41eea1db27540339fd Mon Sep 17 00:00:00 2001
From: Dave Stevenson <dave.stevenson@raspberrypi.org>
Date: Wed, 7 Aug 2019 11:31:08 +0100
Subject: [PATCH] drm/vc4: Add missing NULL check to
 vc4_crtc_consume_event

vc4_crtc_consume_event wasn't checking crtc->state->event was
set before dereferencing it, leading to an OOPS.

Fixes "a5b534b drm/vc4: Resolve the vblank warnings on mode switching"

Signed-off-by: Dave Stevenson <dave.stevenson@raspberrypi.org>
---
 drivers/gpu/drm/vc4/vc4_firmware_kms.c | 3 +++
 1 file changed, 3 insertions(+)

--- a/drivers/gpu/drm/vc4/vc4_firmware_kms.c
+++ b/drivers/gpu/drm/vc4/vc4_firmware_kms.c
@@ -1000,6 +1000,9 @@ static void vc4_crtc_consume_event(struc
 	struct drm_device *dev = crtc->dev;
 	unsigned long flags;
 
+	if (!crtc->state->event)
+		return;
+
 	crtc->state->event->pipe = drm_crtc_index(crtc);
 
 	WARN_ON(drm_crtc_vblank_get(crtc) != 0);
